src/hooks/useAffiliatesInfo.ts

```diff
@@ -38,11 +50,31 @@ export const useAffiliatesInfo = (dydxAddress?: string) => {
     }
   };
 
-  const query = useQuery({
+  const affiliateMetadataQuery = useQuery({
     queryKey: ['affiliatesMetadata', dydxAddress],
-    queryFn,
+    queryFn: fetchAffiliateMetadata,
     enabled: Boolean(compositeClient && dydxAddress),
   });
 
-  return query;
+  const fetchAffiliateMaxEarning = async () => {
+    const taker3FeeTier = feeTiers?.[2].taker ?? DEFAULT_TAKER_3_FEE;
+    const allAffiliateTiers = await getAllAffiliateTiers();
+    let maxRevshare = DEFAULT_MAX_AFFILIATE_SHARE;
+    if (allAffiliateTiers?.tiers?.tiers?.length) {
+      const lastTier = allAffiliateTiers.tiers.tiers[allAffiliateTiers.tiers.tiers.length - 1];
+      maxRevshare = lastTier.takerFeeSharePpm / 1_000_000;
+    }
+
+    const maxEarning = taker3FeeTier * maxRevshare * REF_SHARE_VOLUME_CAP;
+    const maxVipEarning = taker3FeeTier * MAX_AFFILIATE_VIP_SHARE * REF_SHARE_VOLUME_CAP;
+    return { maxEarning, maxVipEarning };
+  };
+
+  const affiliateMaxEarningQuery = useQuery({
+    queryKey: ['affiliateMaxEarning', compositeClient],
```

Comment:
since `feeTiers` is a dependency in the queryFn, you may want to include it in the queryKey here?

Reply:
good point!